tests/ert_tests/lsf_queue/test_lsf_driver.py

```diff
@@ -0,0 +1,128 @@
+    shutil.copytree(
+        os.path.join(source_root, "tests", "ert_tests", "lsf_queue", "mocked_commands"),
+        os.path.join(str(tmpdir), "poly_example", "mocked_commands"),
+    )
+
+    with tmpdir.as_cwd():
+        yield
+
+
+def test_run_mocked_lsf_queue(poly_case_context):
+    apply_customized_config(log_level="WARNING")
+    parser = ArgumentParser(prog="test_main")
+    parsed = ert_parser(
+        parser,
+        [
+            ENSEMBLE_EXPERIMENT_MODE,
+            "poly_example/poly.ert",
+        ],
+    )
+
+    run_cli(parsed)
+
+
+def test_mock_bsub_fail_random(poly_case_context):
+    """
+    Approx 50% of the submits will fail due to the random generator in the
+    created mocked bsub script. By using the retry functionality towards
+    queue-errors in job_queue.cpp we should still manage to finalize all our runs
+    before exhausting the limits
+    """
+    bsub_random_fail_name = "bsub_random_fail"
+    introduce_bsub_failures(
+        script_name=bsub_random_fail_name, percentage_successful_submits=50
+    )
+
+    apply_customized_config(mocked_bsub=bsub_random_fail_name, log_level="WARNING")
+
+    parser = ArgumentParser(prog="test_main")
+    parsed = ert_parser(
+        parser,
+        [
+            ENSEMBLE_EXPERIMENT_MODE,
+            "poly_example/poly.ert",
+        ],
+    )
+
+    run_cli(parsed)
+
+    assert True is True
+
+
+def apply_customized_config(
+    max_running: int = 10,
+    num_realizations: int = 10,
+    min_realizations: int = 1,
+    log_level: str = "INFO",
+    mocked_bsub: str = "mock_bsub",
+):
+
+    # Overwriting the "poly.ert" config file in tmpdir runpath
+    # with our own customized config with at least sets queue option to LSF and
+    # introducing the mocked jobs from "mocked_commands" folder.
+
+    config = [
+        "JOBNAME poly_%d\n",
+        "QUEUE_SYSTEM  LSF\n",
+        f"QUEUE_OPTION LSF  MAX_RUNNING {max_running}\n",
+        "QUEUE_OPTION  LSF  BJOBS_CMD   mocked_commands/mock_bjobs\n",
+        f"QUEUE_OPTION LSF  BSUB_CMD    mocked_commands/{mocked_bsub}\n",
+        "RUNPATH poly_out/real_%d/iter_%d\n",
+        "OBS_CONFIG observations\n",
+        "TIME_MAP time_map\n",
+        f"NUM_REALIZATIONS {num_realizations}\n",
+        f"MIN_REALIZATIONS {min_realizations}\n",
+        "GEN_KW COEFFS coeff.tmpl coeffs.json coeff_priors\n",
+        "GEN_DATA POLY_RES RESULT_FILE:poly_%d.out REPORT_STEPS:0 INPUT_FORMAT:ASCII\n",
+        "INSTALL_JOB poly_eval POLY_EVAL\n",
+        "SIMULATION_JOB poly_eval\n",
+        f"LOG_LEVEL {log_level}\n",
+    ]
+    with open("poly_example/poly.ert", "w") as fh:
+        fh.writelines(config)
+
+
+def introduce_bsub_failures(script_name: str, percentage_successful_submits: int = 100):
+    script = dedent(
+        f"""#!/usr/bin/env python3
+import random
+import sys
+import subprocess
+
+num = random.random()
+if num < {percentage_successful_submits / 100}:
```

Comment:
I also believe that the sign of comparison is the wrong way here. It has no effect since we only use it with `50`, but try injecting `100` here. Which we would expect make it always successful ;)